STM32/SPI/Src/ADS1120.c

```diff
@@ -96,13 +96,56 @@ static bool isDataReady(ADS1120Device *dev)
     return !stmGetGpio(dev->drdy);
 }
 
-//static void powerDown(SPI_HandleTypeDef *hspi);     // Enter power-down mode (not implemented)
+// it is possible to a single register but to simplify stuff, just read it all.
+static HAL_StatusTypeDef readRegisters(ADS1120Device *dev, ADS1120_RegConfig* regs)
+{
+    ADS1120Cmd cmd = { .byte = 0b00100011 };
+
+    HAL_StatusTypeDef ret = HAL_SPI_Transmit(dev->hspi, &cmd.byte, 1, ADS_TIMEOUT);
+    if (ret != HAL_OK)
+        return ret;
+    return HAL_SPI_Receive(dev->hspi, regs->regs, sizeof(regs->regs), ADS_TIMEOUT);
+}
+
+static HAL_StatusTypeDef verifyTransmission(ADS1120Device *dev, ADS1120_RegConfig *result)
+{
+    // Read the current registry setup
+    ADS1120_RegConfig currentReg;
+    bzero(&currentReg, sizeof(currentReg));
```

Comment:
Where is this function coming from? It isn't standard library.
Whats wrong with memset?

Reply:
I reused bzero from the previous implementation where registry settings were compared and has worked up until now - so didn't think much of it. 

I looked a bit into bzero vs. memset and since bzero has been deprecated (a long time ago!) I'll just replace these calls with memset.